### PR TITLE
Add batch input item shape specification

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

File: protobuf/model_config.proto [hunk 1]
```diff
@@ -509,6 +509,28 @@ message BatchInput
...
+    //@@         with shape [2, 2] are batched, the batch input will
+    //@@         have shape [3, 2] and value [ [3, 1], [3, 1], [2, 2]].
+    //@@
+    BATCH_ITEM_SHAPE = 4;
+
+    //@@      .. cpp:enumerator:: Kind::BATCH_ITEM_SHAPE_FLATTEN = 5
+    //@@
+    //@@         Among the requests in the batch, the shape of the
+    //@@         'source_input' will be added as input with shape
+    //@@         [batch_size * len(input_dim)]. For example, if one
```
Comment (CoderHam) [inline protobuf/model_config.proto:528]: Fix to mention flat structure of shape

**Review by CoderHam — CHANGES_REQUESTED** [review]
Both descriptions for BATCH_ITEM_SHAPE and BATCH_ITEM_SHAPE_FLATTEN are identical except in the flatten case the values with be flat. Need to clarify that the values in the second case are flattened

File: protobuf/model_config.proto [hunk 2]
```diff
@@ -509,6 +509,28 @@ message BatchInput
     //@@         The data of the tensor will be uninitialized.
```
Comment (CoderHam) [inline protobuf/model_config.proto:509]: Update copyright?

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: protobuf/model_config.proto (hunk 1 above)
Comment (GuanLuo) [reply protobuf/model_config.proto:528]: Can you clarify this comment?

Comment (tanmayv25) [reply protobuf/model_config.proto:528]: I think we must explicitly call out single dimensionality of tensor. It is hard to ascertain that with just [batch_size * len(input_dim)]

**Review by CoderHam — APPROVED** [review]

**Pull request merged**